pkg/proxy/iptables/proxier.go

```diff
@@ -1343,7 +1343,7 @@ func (proxier *Proxier) syncProxyRules() {
 					// This is very low impact. The NodePort range is intentionally obscure, and unlikely to actually collide with real Services.
 					// This only affects UDP connections, which are not common.
 					// See issue: https://github.com/kubernetes/kubernetes/issues/49881
-					isIPv6 := svcInfo.clusterIP.To4() != nil
+					isIPv6 := utilproxy.IsIPv6(svcInfo.clusterIP.String())
```

Comment:
Rather than convert to string and back, can we make `IsIPv6(ip net.IP)` and `IsIPV6String(ip string)` ?

Reply:
Sounds fair. It's fixed now, please take a look :)